java-checks/src/main/java/org/sonar/java/checks/AnonymousClassShouldBeLambdaCheck.java

```diff
@@ -70,20 +75,51 @@ public void visitNewClass(NewClassTree tree) {
     if (classBody != null) {
       TypeTree identifier = tree.identifier();
       if (!useThisIdentifier(classBody) && !enumConstants.contains(identifier) && isSAM(classBody)) {
-        context.reportIssue(this, identifier, "Make this anonymous inner class a lambda" + context.getJavaVersion().java8CompatibilityMessage());
+        Tree member = classBody.members().get(0);
+        if (member.is(Tree.Kind.METHOD) && multipleMethodsDefaultSameSignature(((JavaSymbol.MethodJavaSymbol) ((MethodTree) member).symbol()))) {
```

Comment:
Name of method is wrong. if you have "_multiple default methods with the same signature_", you DONT want to raise an issue. you condition should be the opposite: `! multipleDefaultMethodsWithSameSignature(...)`  

(Note the change of name of the method: `multipleDefaultMethodsWithSameSignature`)

I also think that this verification should be part of `isSAM()` method, as a last condition. Because that's the essence of a "_Single Abstract Method_" interface. It seems odd to me to answer "true" and then to have to check if it's really a SAM or not.